### DISCUSSION TITLE
Keycloak 19.0.2 does not properly work on firefox

### DISCUSSION THREAD

**Discussion by sirishkumar** [keycloak/keycloak#14923] [Archive]
I have migrated Keycloak running on Kubernetes from 15.1.1. to 19.0.2. Everything works on Brave and Chrome but on Firefox login works, but as soon as  I select Realm settings, it seems broken as shown below and after this error, I get the same error on all the pages.  If I don't select Realm settings, everything seems to be fine.

<img width="662" alt="image" src="https://user-images.githubusercontent.com/27522/196136540-d53bcd1f-c291-4672-b381-de95310ff085.png">

I tried cleaning the Browser cache and deleting all Cookies, but it did not help. 

Issues observed in the Firefox developer console

<img width="1271" alt="image" src="https://user-images.githubusercontent.com/27522/196139003-6e3866b4-3dbf-4ffc-b31a-523e56f6c402.png">

Java script call stack in Developer Console 

```
SyntaxError: missing = in const declaration index.4ad08902.js:25:73
    q3 https://auth.test.our-company.net/auth/resources/yå6i4/admin/keycloak.v2/assets/index.4ad08902.js:25
    payload https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:25
    vt https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:23
    aa https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:23
    Ea https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:27
    Ma https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:27
    o5 https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:27
    Tt https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:27
    o6 https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:25
    z0 https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:23
    unstable_runWithPriority https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:19
    he https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:23
    z0 https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:23
    q https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:19
    onmessage https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:19
    (Async: EventHandlerNonNull)
    js https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:19
    __commonJS https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:1
    js https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:19
    __commonJS https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:1
    js https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:19
    __commonJS https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:1
    js https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:27
    __commonJS https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:1
    <anonymous> https://auth.test.our-company.net/auth/resources/yb6i4/admin/keycloak.v2/assets/index.4ad08902.js:81
```

**Comment by ssilvert** [comment]
@sirishkumar I believe this is a duplicate of https://github.com/keycloak/keycloak-ui/issues/3518. 

It may be fixed in the nightly build.  Can you check it out and let us know if it solves your problem?
https://github.com/keycloak/keycloak/releases/tag/nightly

**Reply by sirishkumar** [reply]
I will try this today. But today I observed it is working fine in the Private window in Firefox with the same warning about KEYCLOAK_3P_COOKIE.

**Reply by jonkoops** [reply]
Pretty sure this is unrelated to the cookie situation. The problem is likely coming from someplace in the JavaScript.

**Reply by sirishkumar** [reply]
@jonkoops  @ssilvert  I tried it again with 19.0.2 and 19.0.3, this issue is not reproducible locally when I run them using docker-compose. Our deployment is on Kubernetes with ISTIO, so we run the Keycloak only in HTTP mode with istio proxy serving as SSL termination. I am not sure why it without any issue with Chrome and Brave but fails in Firefox. Let me know if you need more info

**Reply by jonkoops** [reply]
That is very strange, it almost sounds like the proxy is doing something to the files. But that seems very unlikely.
